### PR TITLE
Make related BC data a single class member

### PR DESCRIPTION
**Pull request by akva2**
Pack all BC data in a single struct, instead of having 6 (12) separate vector members.

Cleaner organization imo.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -2874,22 +2863,22 @@ private:
...
+                        data = &freebc_.plus[1];
                         break;
                     case FaceDir::ZMinus:
-                        data = &freebcZMinus_;
+                        data = &freebc_.minus[2];
                         break;
                     case FaceDir::ZPlus:
-                        data = &freebcZ_;
+                        data = &freebc_.plus[2];
                         break;
```
Comment (blattms) [inline ebos/eclproblem.hh:2882]: Why not get rid of this switch-case by using a container for both plus and minus that allows lookup with `bcface.dir`like `data = freebc.data[bcface_dir]`. But maybe that is not worth it as there are other swich-cases that won't be affected.

**Review by blattms — APPROVED** [review]
Just one minor/silly question

File: ebos/eclproblem.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclproblem.hh:2882]: i did consider it, but found it too much for too little gain. but i can do it if you insist.

Comment (blattms) [reply ebos/eclproblem.hh:2882]: Nope. I have the same impression now.

**Comment by akva2** [comment]
okay, I pushed a little harder. I think this version is better.

**Comment by akva2** [comment]
jenkins build this opm-common=3180 please

File: ebos/eclproblem.hh [hunk 2]
```diff
@@ -3018,20 +2914,35 @@ private:
...
+        {
+            for (auto& d : data)
+                d.resize(size, defVal);
+        }
+
+        const std::vector<T>& operator()(FaceDir::DirEnum dir) const
+        {
+            if (dir == FaceDir::DirEnum::Unknown)
+                throw std::runtime_error("Tried to acess BC data for the 'Unknown' direction");
+            int idx = 31 - __builtin_clz(static_cast<int>(dir));
```
Comment (bska) [inline ebos/eclproblem.hh:2932]: I **really** don't think we should use compiler intrinsics.  I'd prefer we use something more portable&mdash;e.g., our home-grown [`count_leading_zeros()`](https://github.com/OPM/opm-common/blob/f414cdd30e5792b1198cfe08082a7605b28668b8/src/opm/input/eclipse/Schedule/WriteRestartFileEvents.cpp#L61) function or its 32-bit equivalent.

Comment (akva2) [reply ebos/eclproblem.hh:2932]: if that's a concern i'd rather just use the 2 line while loop. okay?

**Comment by akva2** [comment]
jenkins build this opm-common=3180 please

**Review by bska — APPROVED** [review]
Thanks.  This is a much welcome simplification.  I'll merge into master.

**Pull request merged**